trino/auth.py

```diff
@@ -421,10 +420,21 @@ def _attempt_oauth(self, response: Response, **kwargs: Any) -> None:
         if not _OAuth2TokenBearer._BEARER_PREFIX.search(auth_info):
             raise exceptions.TrinoAuthError(f"Error: header info didn't match {auth_info}")
 
-        auth_info_headers = parse_dict_header(
-            _OAuth2TokenBearer._BEARER_PREFIX.sub("", auth_info, count=1))  # type: ignore
+        split_challenge = auth_info.split(" ", 1)
+        self.scheme = split_challenge[0]
+        trimmed_challenge = split_challenge[1] if len(split_challenge) > 1 else ""
 
-        auth_server = auth_info_headers.get('x_redirect_server')
+        auth_info_headers = {}
+
+        for item in trimmed_challenge.split(","):
+            comps = item.split("=")
+            if len(comps) == 2:
+                key = comps[0].strip(' "')
+                value = comps[1].strip(' "')
+                if key:
+                    auth_info_headers[key.lower()] = value
```

Comment:
the dict would overwrite if same key is present (e.g. if there are multiple `Bearer realm` entries or multiple `Bearer x_redirect_server`).

Reply:
I don't think it's possible based on https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate

Reply:
The entire challenge cannot be a duplicate I think but here the `key` is the part split on `=`.

So the example at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#digest_authentication_with_sha-256_and_md5 has two challenges for which `key = 'Digest realm'`.

(Sorry I'm AFK and did not test the static method against that example 🙏).